File: 23/template.ts
import assert from "assert";
import { getContent } from "./utils";

const TEST = 0;
const FILENAME = __filename.split(".")[0] + (TEST ? "-test.txt" : "-input.txt");

const prepareInput = (data: string) => {
  const lines = data.split("\n").map((l) => l.trim());
  return lines;
};

const data = prepareInput(getContent(FILENAME));

// --- PART1 ---
const RESULT_ONE = 0;

const part1 = (data: any) => {
  return data;
};

const result1 = part1(data);
TEST && assert.equal(result1, RESULT_ONE, "RESULT 1 INCORRECT");
console.log("1:", result1);

// --- PART 2 ---
const RESULT_TWO = 0;

const part2 = (data: any) => {
  return data;
};

const result2 = part2(data);
TEST && assert.equal(result2, RESULT_TWO, "RESULT 2 INCORRECT");
console.log("2:", result2);

File: 23/template.c
#include "utils.h"
#include <assert.h>
#include <stdio.h>
#include <stdlib.h>

#define TEST 1
#define NUM 01

int part_1(char **lines);

int part_2(char **lines);

int main(int argc, char *argv[]) {
  char *template = TEST ? "e%02d-test.txt" : "e%02d-input.txt";
  char filename[18];
  sprintf(filename, template, NUM);

  FILE *f = fopen(filename, "r");
  assert(f != NULL);

  char **lines = NULL;
  int lineCount = read_lines(&lines, f);
  assert(lines != NULL);
  fclose(f);

  for (int i = 0; i < lineCount; i++) {
    char *line = lines[i];
    fputs(line, stdout);
  }

  int part_1_result = 0;
  int part_2_result = 0;

  printf("PART 1: %d\n", part_1_result);
  printf("PART 2: %d\n", part_2_result);

  return 0;
}
